src/app/zones/data/index.ts

```diff
@@ -1,36 +1,87 @@
-import { getIsConnected } from '@/app/subscriptions/data'
-import type { ZoneOverview } from '@/types/index.d'
+import type { PaginatedApiListResponse as CollectionResponse } from '@/types/api.d'
+import type {
+  ZoneOverview as PartialZoneOverview,
+  ZoneInsight as PartialZoneInsight,
+  Zone as PartialZone,
+  KDSSubscription,
+} from '@/types/index.d'
 import { get } from '@/utilities/get'
 
-// TODO(jc): These two are extremely similar, see if we can merge
-export function getZoneControlPlaneEnvironment(zoneOverview: ZoneOverview): string {
-  // TODO(jc): is it ok to use the config from the first subscription we find here?
-  for (const subscription of zoneOverview.zoneInsight?.subscriptions ?? []) {
-    if (subscription.config) {
-      return JSON.parse(subscription.config).environment
-    }
-  }
-  return ''
+export type Zone = PartialZone & {
+  enabled: boolean
+}
+export type ZoneInsight = PartialZoneInsight & {
+  connectedSubscription?: KDSSubscription
+  config: Record<string, unknown>
+  authenticationType: string
+  environment: string
+}
+export type ZoneOverview = PartialZoneOverview & {
+  zoneInsight?: ZoneInsight
+  zone: Zone
+  state: 'online' | 'offline' | 'disabled'
 }
-export function getZoneDpServerAuthType(zone: ZoneOverview): string {
-  const subscriptions = zone.zoneInsight?.subscriptions ?? []
-  if (subscriptions.length > 0) {
-    const lastSubscription = subscriptions[subscriptions.length - 1]
-    if (lastSubscription.config) {
-      const parsedConfig = JSON.parse(lastSubscription.config)
-      return get(parsedConfig, 'dpServer.auth.type', '')
+
+export const Zone = {
+  fromObject: (item: PartialZone): Zone => {
+    return {
+      ...item,
+      enabled: !(item.enabled === false),
     }
-  }
-  return ''
+  },
 }
-// TODO(jc): end
+export const ZoneInsight = {
+  fromObject: (item?: PartialZoneInsight): ZoneInsight | undefined => {
+    // if item isn't set don't even try augmenting things
+    return isSet<PartialZoneInsight>(item)
+      ? ((item) => {
+        // figure out the connectedSubscription by looking at the connectTime
+        // and disconnectTime of the last subscription
+        const connectedSubscription: KDSSubscription | undefined = !Array.isArray(item.subscriptions)
+          ? undefined
+          : [item.subscriptions[item.subscriptions.length - 1]].find((item) => item.connectTime?.length && !item.disconnectTime)
+
+        // using the connectedSubscription find the config for the zone if it exists, is valid JSON and is not null and
+        // turn it into an object
+        const config: Record<string, unknown> = (() => {
+          const str = isSet<string>(connectedSubscription?.config) ? connectedSubscription.config : '{}'
+          try {
+            return JSON.parse(str)
+          } catch (e) { }
+          return {}
+        })()
```

Comment:
**Issue**: We shouldn’t expect the possibility that `JSON.parse(str)` can throw. Instead, we should rely on the assumption that, if the ZoneInsight was updated following an applied config, that that config was parseable in the first place. If not, I’d assume the config to not make it into a KDS subscription.

On top of that, hiding the error here is dangerous because we won’t know that the error occurred. Much better to let this cause an exception that we can actually fix (because it’s not an expected application scenario).

```suggestion
        const config: Record<string, unknown> = isSet<string>(connectedSubscription?.config) ? JSON.parse(connectedSubscription.config) : {}
```

Reply:
I added a `console.error` here which I think addresses your concern about us not knowing if this throws, but I guess we shouldn't expect that. But if it does happen, `config` would just be a `{}` and things would continue on as normal.

Reply:
I'm guessing I can resolve this, let me know if not